galsim/config/output.py

```diff
@@ -467,22 +500,47 @@ def getNImages(self, config, base, file_num):
         """
         return 1
 
-    def writeFile(self, data, file_name):
+    def canAddHdus(self):
+        """Returns whether it is permissible to add extra HDUs to the end of the data list.
+
+        In the base class, this returns True.
+        """
+        return True
```

Comment:
The DataCube output type cannot add additional HDUs to the end, since it collapses all the images into a single 3d array.  One could probably write a version of that which smartly handled extra weight and badpix hdus in the data list and made data cubes for them as well, but the current implementation doesn't do so.